plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,352 @@
+                } catch (RuntimeException e) {
+                    reTryTimes = result.getReTryTimes();
+                    reTryTimes++;
+                    result.setReTryTimes(reTryTimes);
+                    log.error("Can't auto reload model in node id {} ,has tried {} times\nThe reason is:{}", localNodeId, reTryTimes, e);
+                }
+
+                // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+                saveLatestReTryTimes(
+                    localNodeId,
+                    reTryTimes,
+                    ActionListener.wrap(saveLatestReTryTimesStep::onResponse, saveLatestReTryTimesStep::onFailure)
+                );
+            }, getReTryTimesStep::onFailure);
+        }, queryTaskStep::onFailure);
+
+        saveLatestReTryTimesStep.whenComplete(response -> log.info("successfully complete all steps"), saveLatestReTryTimesStep::onFailure);
+    }
+
+    /**
+     * auto reload 1 model under the node id
+     *
+     * @param localNodeId node id
+     * @param modelId     model id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeAndModelId(String localNodeId, String modelId) throws RuntimeException {
+        String[] allNodeIds = nodeHelper.getAllNodeIds();
+        List<String> allNodeIdList = new ArrayList<>(List.of(allNodeIds));
+        if (!allNodeIdList.contains(localNodeId)) {
+            allNodeIdList.add(localNodeId);
+        }
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, allNodeIdList.toArray(new String[] {}), false, false);
+
+        client
+            .execute(
+                MLLoadModelAction.INSTANCE,
+                mlLoadModelRequest,
+                ActionListener
+                    .wrap(response -> log.info("the model {} is auto reloading under the node {} ", modelId, localNodeId), exception -> {
+                        log.error("fail to reload model " + modelId + " under the node " + localNodeId + "\nthe reason is: " + exception);
+                        throw new RuntimeException(
+                            "fail to reload model " + modelId + " under the node " + localNodeId + "\nthe reason is: " + exception
+                        );
+                    })
+            );
+    }
+
+    /**
+     * query task index, and get the result of "task_type"="LOAD_MODEL" and "state"="COMPLETED" and
+     * "worker_node" match nodeId
+     *
+     * @param localNodeId one of query condition
+     */
+    @VisibleForTesting
+    void queryTask(String localNodeId, ActionListener<SearchResponse> searchResponseActionListener) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().from(0).size(1);
+
+        QueryBuilder queryBuilder = QueryBuilders
+            .boolQuery()
+            .must(QueryBuilders.matchPhraseQuery("task_type", "LOAD_MODEL"))
+            .must(QueryBuilders.matchPhraseQuery("state", "COMPLETED"))
```

Comment:
I think we should also consider `COMPLETED_WITH_ERROR` which means the model isn't loaded to all workers nodes, but loaded on some worker nodes successfully.

Reply:
Okay, I found that there may be a time difference, when I wrote the code，the  enumeration class `MLTaskState` does not exist `COMPLETED_WITH_ERROR`, I add this to my code